python/cudf/cudf/pylibcudf_tests/test_column_factories.py

```diff
@@ -0,0 +1,266 @@
+    pa.uint8(),
+    pa.uint16(),
+    pa.uint32(),
+    pa.uint64(),
+    pa.int8(),
+    pa.int16(),
+    pa.int32(),
+    pa.int64(),
+    pa.float32(),
+    pa.float64(),
+    pa.bool_(),
+]
+
+TIMESTAMP_TYPES = [
+    pa.timestamp("s"),
+    pa.timestamp("ms"),
+    pa.timestamp("us"),
+    pa.timestamp("ns"),
+]
+
+DURATION_TYPES = [
+    pa.duration("s"),
+    pa.duration("ms"),
+    pa.duration("us"),
+    pa.duration("ns"),
+]
+
+DECIMAL_TYPES = [pa.decimal128(38, 2)]
+
+STRING_TYPES = [pa.string()]
+STRUCT_TYPES = [DEFAULT_STRUCT_TESTING_TYPE]
+LIST_TYPES = [pa.list_(pa.int64())]
+
+ALL_TYPES = (
+    NUMERIC_TYPES
+    + TIMESTAMP_TYPES
+    + DURATION_TYPES
+    + STRING_TYPES
+    + DECIMAL_TYPES
+    + STRUCT_TYPES
+    + LIST_TYPES
+)
+
+
+def pa_type_to_plc_type(pa_type):
+    # TODO: should be a cleaner way
+    return plc.interop.from_arrow(pa.array([], type=pa_type)).type()
+
+
+@pytest.fixture(scope="module", params=NUMERIC_TYPES, ids=repr)
+def numeric_pa_type(request):
+    return request.param
+
+
+@pytest.fixture(
+    scope="module",
+    params=DECIMAL_TYPES,
+    ids=repr,
+)
+def fixed_point_pa_type(request):
+    return request.param
+
+
+@pytest.fixture(
+    scope="module",
+    params=TIMESTAMP_TYPES,
+    ids=repr,
+)
+def timestamp_pa_type(request):
+    return request.param
+
+
+@pytest.fixture(
+    scope="module",
+    params=DURATION_TYPES,
+    ids=repr,
+)
+def duration_pa_type(request):
+    return request.param
+
+
+@pytest.fixture(
+    scope="module",
+    params=[
+        plc.MaskState.UNALLOCATED,
+        plc.MaskState.ALL_VALID,
+        plc.MaskState.ALL_NULL,
+        plc.MaskState.UNINITIALIZED,
+    ],
+    ids=["unallocated", "all_valid", "all_null", "uninitialized"],
+)
+def mask_state(request):
+    return request.param
+
+
+def test_make_empty_column_dtype(pa_type):
+    pa_col = pa.array([], type=pa_type)
+
+    # TODO: DataType.from_arrow()?
+    plc_type = plc.interop.from_arrow(pa_col).type()
+
+    if isinstance(pa_type, (pa.ListType, pa.StructType)):
+        with pytest.raises(ValueError):
+            plc.column_factories.make_empty_column(plc_type)
+        return
+
+    cudf_col = plc.column_factories.make_empty_column(plc_type)
+    assert_column_eq(cudf_col, pa_col)
+
+
+def test_make_empty_column_typeid(pa_type):
+    pa_col = pa.array([], type=pa_type)
+
+    # TODO: DataType.from_arrow()?
+    tid = plc.interop.from_arrow(pa_col).type().id()
+
+    if isinstance(pa_type, (pa.ListType, pa.StructType)):
+        with pytest.raises(ValueError):
+            plc.column_factories.make_empty_column(tid)
+        return
+
+    cudf_col = plc.column_factories.make_empty_column(tid)
+    assert_column_eq(cudf_col, pa_col)
+
+
+def test_make_numeric_column(numeric_pa_type, mask_state):
+    if mask_state == plc.column_factories.MaskState.ALL_NULL:
+        expected = pa.array([None] * size, type=numeric_pa_type)
+    else:
+        # TODO: uninitialized not necessarily 0
```

Comment:
What does this mean?

Reply:
I absolutely don't want to initialize in pylibcudf if the matching libcudf function doesn't initialize, so either zero-fill in the test, or don't check values.

Reply:
https://github.com/rapidsai/cudf/pull/15257/commits/10b07b81cb183c1aa798d3edc6ffd762b8d5d3c6 introduces a way of validating the empty column without comparing it to a pyarrow array directly